indico/modules/events/registration/google_wallet.py

```diff
@@ -0,0 +1,265 @@
+            logo_url = config.LOGO_URL if 'http' in config.LOGO_URL else f'{config.BASE_URL}{config.LOGO_URL}'
+        dict_template = {
+            'id': f'{issuer_id}.{class_suffix}',
+            'issuerName': self.settings['google_wallet_issuer_name'],
+            'reviewStatus': 'UNDER_REVIEW',
+            'eventName': {
+                'defaultValue': {
+                    'language': 'en-US',
+                    'value': self.event.title
+                }
+            },
+            'venue': {
+                'name': {
+                    'defaultValue': {
+                        'language': 'en-US',
+                        'value': self.event.venue_name if not self.event.room_name
+                        else f'{self.event.venue_name} - Room {self.event.room_name}'
+                    }
+                },
+                'address': {
+                    'defaultValue': {
+                        'language': 'en-US',
+                        'value': self.event.address
+                    }
+                }
+            },
+            'logo': {
+                'sourceUri': {
+                    'uri': logo_url
+                }
+            },
+            'textModulesData': [
+                {
+                    'header': 'Date',
+                    'language': 'en-US',
+                    'body': make_format_event_date_func(self.event.category)(self.event),
+                    'id': 'datefield'
+                },
+            ],
+            'classTemplateInfo': {
+                'cardTemplateOverride': {
+                    'cardRowTemplateInfos': [{
+                        'oneItem': {
+                            'item': {
+                                'firstValue': {
+                                    'fields': [{
+                                        'fieldPath': "class.textModulesData['datefield']"
+                                    }]
+                                }
+                            },
+                        }
+                    }, {
+                        'twoItems': {
+                            'startItem': {
+                                'firstValue': {
+                                    'fields': [{
+                                        'fieldPath': "object.textModulesData['namefield']"
+                                    }]
+                                }
+                            },
+                            'endItem': {
+                                'firstValue': {
+                                    'fields': [{
+                                        'fieldPath': "object.textModulesData['emailfield']"
+                                    }]
+                                }
+                            },
+                        }
+                    }]
+                }
+            }
+        }
+        if not dict_template['venue']['name']['defaultValue']['value']:
+            dict_template.pop('venue')
```

Comment:
Wouldn't it be cleaner to add the venue only if you have a value?

Reply:
OK, I'll do it but for reference, if there is not venue/address, the relative props are set to '' (event.venue_name, event.room_name, event.address) and in the Ticket generation, when the value is empty string, it is not added by default.